### PR TITLE
centralize sc builder data to minimize io ops

### PR DESCRIPTION
**Pull request by carlitosan**
Cache the results of parsing graphs for editor component data in a system component so that it can be re-used across multiple editor components and prefabs during play in editor and prefab compilation.

### CODE REVIEW

File: Gems/ScriptCanvas/Code/Editor/SystemComponent.h [hunk 1]
```diff
@@ -14,15 +14,18 @@
...
 #include <AzToolsFramework/AssetBrowser/AssetBrowserBus.h>
 #include <AzToolsFramework/Editor/EditorContextMenuBus.h>
+#include <AzToolsFramework/Entity/EditorEntityContextBus.h>
+#include <Builder/ScriptCanvasBuilder.h>
 #include <Core/GraphBus.h>
 #include <Editor/View/Windows/Tools/UpgradeTool/Model.h>
 #include <ScriptCanvas/Bus/ScriptCanvasBus.h>
 #include <ScriptCanvas/Bus/ScriptCanvasExecutionBus.h>
-#include <AzToolsFramework/Entity/EditorEntityContextBus.h>
+#include <Builder/ScriptCanvasBuilderDataSystem.h>
```
Comment (lsemp3d) [inline Gems/ScriptCanvas/Code/Editor/SystemComponent.h:28]: nit: can you sort these?

**Review by lsemp3d — APPROVED** [review]

File: Gems/ScriptCanvas/Code/Editor/SystemComponent.h (hunk 1 above)
Comment (carlitosan) [reply Gems/ScriptCanvas/Code/Editor/SystemComponent.h:28]: that is never a nit. After the current PRs close, I'm doing a full sort pass on a couple of files that are really disorganized.

**Review by cgalvan — APPROVED** [review]

**Review by onecent1101 — COMMENTED** [review]
Can you provide more information in PR details to explain how it worked before?
I can see that we are trying to store build results in memory in this PR, so before this change all build results are generated on the fly?

**Comment by lsemp3d** [comment]
> Can you provide more information in PR details to explain how it worked before? I can see that we are trying to store build results in memory in this PR, so before this change all build results are generated on the fly?

+1 to this request 👍🏼

**Pull request merged**